extensions/wikia/AdEngine/js/template/modal.js

```diff
@@ -50,9 +92,25 @@ define('ext.wikia.adEngine.template.modal', [
 				width: params.width,
 				classes: 'wikia-ad-iframe'
 			}),
+			async = params.canHop && params.slotName,
+			gptEventMock = {
+				size: {
+					width: params.width,
+					height: params.height
+				}
+			},
+			modalHandler,
 			skin = adContext.getContext().targeting.skin,
 			lightboxParams = lightBoxExpansionModel[skin];
 
+		if (skin === 'mercury') {
```

Comment:
:+1: great idea, I've created modalHandlerFactory and modules for oasis and mercury